oversweet-infra/src/main/java/com/depromeet/oversweet/redis/RedisRepository.java

```diff
@@ -0,0 +1,49 @@
+package com.depromeet.oversweet.redis;
+
+import com.depromeet.oversweet.exception.ErrorCode;
+import com.depromeet.oversweet.exception.redis.OverSweetRedisException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * 레디스의 접근하여 데이터를 저장하거나 가져오는 Repository 역할 클래스
+ */
+@RequiredArgsConstructor
+@Repository
+public class RedisRepository {
+    private final RedisTemplate<String, Object> redisTemplate;
+    private final ObjectMapper objectMapper;
+
+    public <T> List<T> getData(final String key, final Class<T> classType) {
+        final String jsonData = (String) redisTemplate.opsForValue().get(key);
+
+        if (StringUtils.hasText(jsonData)) {
+            try {
+                return objectMapper.readValue(jsonData, objectMapper.getTypeFactory().constructCollectionType(List.class, classType));
+            } catch (Exception e) {
+                throw new OverSweetRedisException(ErrorCode.OVERSWEET_REDIS_JSON_ERROR);
+            }
+        }
+        return new ArrayList<>();
+    }
```

Comment:
StringUtils 보단 Optional을 활용하면 어떨까요??
```suggestion
        public <T> List<T> getData(final String key, final Class<T> classType) {
        return Optional.ofNullable(redisTemplate.opsForValue().get(key))
                .map(jsonData -> {
                    try {
                        return objectMapper.readValue(jsonData, objectMapper.getTypeFactory().constructCollectionType(List.class, classType));
                    } catch (JsonProcessingException e) {
                        throw new OverSweetRedisException(ErrorCode.OVERSWEET_REDIS_JSON_ERROR);
                    }
                })
                .orElseGet(Collections::emptyList);
    }
```

추가적으로 Exception 보단 JsonProcessingException 을 활용하면 어떨까요??